src/memgraph.cpp

```diff
@@ -424,6 +430,45 @@ void InitializeLogger() {
   CreateLoggerFromSink(sinks, ParseLogLevel());
 }
 
+std::pair<std::string, std::string> LoadUsernameAndPassword(std::string pass_file) {
+  std::ifstream file(pass_file);
+  if (file.fail()) {
+    spdlog::warn("Problem with opening MG_PASSFILE, memgraph server will start without user");
+    return {"", ""};
+  }
+  std::vector<std::string> result;
+
+  std::string line;
+  std::getline(file, line);
+  size_t pos = 0;
+  std::string token;
+  std::string delimiter = ":";
+  while ((pos = line.find(delimiter)) != std::string::npos) {
+    if (line[pos - 1] == '\\') {
+      line.erase(pos - 1, 1);
+      token += line.substr(0, pos);
+      line.erase(0, pos);
+
+    } else {
+      token += line.substr(0, pos);
+      result.push_back(token);
+      line.erase(0, pos + delimiter.length());
+      token = "";
+    }
```

Comment:
I am not sure I understand why are we doing this

Reply:
We are doing this because someone can have `:` in their username or password for example ni:ko:krvavica which then would lead to wrong parsing but with this if you put ni\:ko:krvavica you will get right username and password.

Reply:
I think it's because of the `pass file`

Reply:
Aha ok maybe just remove that part somewhere else and possibly rewrite it something like this:
```suggestion
std::size_t ReplaceAllExceptLast(std::string& inout, std::string_view what, std::string_view with)
{
    std::size_t count{};
    auto last_occurance = inout.find_last_of(what);
    for (std::string::size_type pos{};
         inout.npos != (pos = inout.find(what.data(), pos, what.length())) && inout.find_last_of(what) != pos;
         pos += with.length(), ++count) {
        inout.replace(pos, what.length(), with.data(), with.length());
    }
    return count;
}
```
you have a similar example here https://en.cppreference.com/w/cpp/string/basic_string/replace, but honestly, I would just forbid `:` in the username, I think that character makes only sense in the password, but I leave that decision to you